### ISSUE TITLE
Change local filename of downloaded pre-built index

### ISSUE THREAD

**Issue by lintool** [castorini/anserini#2406]
In `~/.cache/pyserini/indexes/`, for pre-built indexes, we have:

```
lucene-index.msmarco-v1-passage.20221004.252b5e
lucene-index.msmarco-v1-passage.20221004.252b5e.c697b18c9a0686ca760583e615dbe450
```

When Pyserini downloads it, the directory name gets postpended with the MD5 checksum; this doesn't happen with Anserini, so we end up downloading two copies of identical files.

@ArthurChen189 can we change the Anserini download name so it gets the MD5 checksum appended, to be consistent with Pyseirni?

**Comment by lintool** [comment]
@16BitNarwhal worked on this: https://github.com/castorini/anserini/pull/2412

**Comment by lintool** [comment]
@16BitNarwhal I don't think the issue has been fixed though... :(

On master, I'm running:

```
java -cp target/anserini-0.24.3-SNAPSHOT-fatjar.jar io.anserini.reproduce.RunMsMarco
```

I'm getting:

```
jimmylin@ubuntu2204-102:~/.cache/pyserini/indexes$ ls -l
total 24054864
-rw-r----- 1 jimmylin jimmylin 24632179766 Mar 21 14:39 lucene-hnsw.msmarco-v1-passage-cos-dpr-distil.20240108.825148.tar.gz
drwxr-x--x 2 jimmylin jimmylin          19 Oct  6  2022 lucene-index.cacm.20221005.252b5e.cfe14d543c6a27f4d742fb2d0099b8e0
drwxr-x--x 2 jimmylin jimmylin          15 Oct  4  2022 lucene-index.msmarco-v1-passage.20221004.252b5e
drwxr-x--x 2 jimmylin jimmylin          15 Oct  4  2022 lucene-index.msmarco-v1-passage.20221004.252b5e.c697b18c9a0686ca760583e615dbe450
drwxr-x--x 2 jimmylin jimmylin          15 May 24  2023 lucene-index.msmarco-v1-passage-splade-pp-ed.20230524.a59610
drwxr-x--x 2 jimmylin jimmylin          15 May 24  2023 lucene-index.msmarco-v1-passage-splade-pp-ed.20230524.a59610.4b3c969033cbd017306df42ce134c395
...
```

E.g., we're getting two copies of the index... (I've cleared the cache before I started...).

**Comment by 16BitNarwhal** [comment]
Just sent a PR for the fix with details!

**Comment by 16BitNarwhal** [comment]
Nevermind, some checks failed 😔

**Comment by lintool** [comment]
Closed by #2413
